toucan_connectors/snowflake/snowflake_connector.py

```diff
@@ -152,44 +171,45 @@ def schema_extra(schema: Dict[str, Any], model: Type['SnowflakeConnector']) -> N
             schema['properties'] = {k: schema['properties'][k] for k in ordered_keys}
 
     @property
-    def access_token(self) -> Optional[str]:
+    def access_token(self) -> str | None:
         return self.user_tokens_keeper and self.user_tokens_keeper.access_token.get_secret_value()
 
     @property
-    def refresh_token(self) -> Optional[str]:
+    def refresh_token(self) -> str | None:
         return self.user_tokens_keeper and self.user_tokens_keeper.refresh_token.get_secret_value()
 
     @property
-    def client_id(self) -> Optional[str]:
+    def client_id(self) -> str | None:
         return self.sso_credentials_keeper and self.sso_credentials_keeper.client_id
 
     @property
-    def client_secret(self) -> Optional[str]:
+    def client_secret(self) -> str | None:
         return (
             self.sso_credentials_keeper
             and self.sso_credentials_keeper.client_secret.get_secret_value()
         )
 
     @staticmethod
-    def _get_status_details(index: int, status: Optional[bool]):
+    def _get_status_details(index: int, status: bool | None) -> list[tuple[str, bool | None]]:
         checks = ['Connection to Snowflake', 'Default warehouse exists']
-        ok_checks = [(check, True) for i, check in enumerate(checks) if i < index]
+        ok_checks: list[tuple[str, bool | None]] = [
```

Comment:
```suggestion
        ok_checks: list[tuple[str, bool]] = [
```

For what  can see, it's always True, never None

Reply:
Yes, but we need a looser type here since were concatenating all checks lists in the end